rest/tests/test_rest_block.py

```diff
@@ -39,6 +39,8 @@ def __init__(self, events):
 
     def poll(self, paging=False):
         self._poll_events[0].set()
+        if len(self._poll_events) == 1:
+            return
```

Comment:
It looks to me like this is just a work-around to get the test to work but it is not really testing the block thoroughly. The block has a retry_limit property and I would expect this test to check that the request is made retry_limit+1 times. This code here is limiting the number of calls to poll instead of letting the block figure it out.